api/PayrollProcessor.Core.Domain/Features/Departments/DepartmentPayrollCreateCommand.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using Ardalis.GuardClauses;
+using PayrollProcessor.Core.Domain.Features.Employees;
+using PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands;
+
+namespace PayrollProcessor.Core.Domain.Features.Departments
+{
+    public class DepartmentPayrollCreateCommand : ICommand<DepartmentPayroll>
+    {
+        public Employee Employee { get; }
+        public EmployeePayroll EmployeePayroll { get; }
+        public Guid RecordId { get; }
```

Comment:
Couldn't think up a good name for this, but I wanted the caller of the command to supply the new `Guid` that will be sent to Cosmos, as opposed to it being generated internally.

This could let us move from returning the new entity directly from the call (`ICommand<DepartmentPayroll>`) to only returning 'success' (`ICommand`) and having more of a disconnected architecture with Commands not returning values.